sklearn/utils/validation.py

```diff
@@ -105,28 +103,20 @@ def _assert_all_finite(
     # false positives from overflow in sum method. The sum is also calculated
     # safely to reduce dtype induced overflows.
     is_float = X.dtype.kind in "fc"
-    if is_float and (np.isfinite(_safe_accumulator_op(np.sum, X))):
+    if is_float and np.isfinite(np.sum(X)):
```

Comment:
The use of `_safe_accumulator_op` is dropped as per https://github.com/scikit-learn/scikit-learn/pull/23197#issuecomment-1115105515, specifically noting that false positives will still be caught in the second-pass while this offers a simplification in the average case.

Reply:
Actually reverted this now since without the `_safe_accumulator_op` this results in overflows for 16-bit data passed to `StandardScalar` 

https://github.com/scikit-learn/scikit-learn/blob/4b252fb126e2c81b1ff1d219905d0e46016f014c/sklearn/preprocessing/tests/test_data.py#L431-L440

Reply:
Looking at commits only, `test_scaler_float16_overflow` was added the same time `_safe_accumlator_op` was: https://github.com/thomasjpfan/scikit-learn/commit/1f5bcaeb39698c9a150ef92cdeaeec75c355a274 so they are connected. Let's keep `_safe_accumlator_op` for now.